### DISCUSSION TITLE
Why Different Instrumentation is there for log4j2.17 and log4j2.7?

### DISCUSSION THREAD

**Discussion by saytoshivam** [open-telemetry/opentelemetry-java-instrumentation#7292] [Q&A]
As i want to add "span name" together with traceId ,spanId etc in logs by overiding log4j instrumentation.So do i need to do it separately for different versions of log4j?The below code snippets is working fine for me , but i have doubt if they will work for all log4j versions.

<img width="1361" alt="image" src="https://user-images.githubusercontent.com/65863999/203706324-9eaf5a70-257e-4f6e-b602-5e40d500cd22.png">

<img width="894" alt="image" src="https://user-images.githubusercontent.com/65863999/203706681-3b49b8e2-371d-408a-b1d6-0ca202f98f2e.png">

<img width="916" alt="image" src="https://user-images.githubusercontent.com/65863999/203706474-0fec93f8-1cac-4bf4-8f10-eb7c60b31029.png">

**Comment by trask** [comment]
hi @saytoshivam, I'd probably recommend creating your own Log4j2 `ContextDataProvider` to implement this, instead of trying to override the javaagent instrumentation
